pkg/aws/commandbuilder.go

```diff
@@ -0,0 +1,126 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type Service string
+
+const (
+	UNDEFINED_SERVICE Service = "undefined service"
+	IAM               Service = "iam"
+)
+
+type Command string
+
+const (
+	UNDEFINED_COMMAND                Command = "undefined command"
+	CREATE_ROLE                      Command = "create-role"
+	DELETE_ROLE                      Command = "delete-role"
+	CREATE_POLICY                    Command = "create-policy"
+	DELETE_POLICY                    Command = "delete-policy"
+	CREATE_POLICY_VERSION            Command = "create-policy-version"
+	DELETE_ROLE_POLICY               Command = "delete-role-policy"
+	ATTACH_ROLE_POLICY               Command = "attach-role-policy"
+	DETACH_ROLE_POLICY               Command = "detach-role-policy"
+	TAG_POLICY                       Command = "tag-policy"
+	TAG_ROLE                         Command = "tag-role"
+	CREATE_OPEN_ID_CONNECT_PROVIDER  Command = "create-open-id-connect-provider"
+	DELETE_OPEN_ID_CONNECT_PROVIDER  Command = "delete-open-id-connect-provider"
+	DELETE_ROLE_PERMISSIONS_BOUNDARY Command = "delete-role-permissions-boundary"
+)
+
+type Param string
+
+const (
+	UNDEFINED_PARAM              Param = "undefined param"
+	TAGS                         Param = "tags"
+	ROLE_NAME                    Param = "role-name"
+	ASSUME_ROLE_POLICY_DOCUMENT  Param = "assume-role-policy-document"
+	PERMISSIONS_BOUNDARY         Param = "permissions-boundary"
+	PATH                         Param = "path"
+	POLICY_NAME                  Param = "policy-name"
+	POLICY_DOCUMENT              Param = "policy-document"
+	POLICY_ARN                   Param = "policy-arn"
+	URL                          Param = "url"
+	CLIENT_ID_LIST               Param = "client-id-list"
+	THUMBPRINT_LIST              Param = "thumbprint-list"
+	OPEN_ID_CONNECT_PROVIDER_ARN Param = "open-id-connect-provider-arn"
+	SET_AS_DEFAULT               Param = "set-as-default"
+)
+
+type ICommandBuilder interface {
+	SetService(awsService Service)
+	SetCommand(awsCommand Command)
+	AddParam(awsParam Param, value interface{})
+	ResetParams()
+	Build() string
+}
```

Comment:
Not atm, but if we ever want to change other commands that are strings, like rosa commands for example I would use the interface to setup the concrete rosa command builder

Reply:
Lets not make a premature interface then. If we ever have multiple implementations for a certain interface we could always create that interface as needed - it might even not include all the current methods.